### DISCUSSION TITLE
[IntelliSense] when using Angular host.class

### DISCUSSION THREAD

**Discussion by leon** [tailwindlabs/tailwindcss#9422] [Ideas]
Since we have started using tailwind in Angular I have switched from using a scss file per component, to not having to have a scss file at all.

And by setting `encapsulation: ViewEncapsulation.None` we let tailwind do all the styling work, which has worked great :)

The only problem I have is that because the `host: { class: '...' }` is within the `@Component` decorator tailwinds auto complete doesn't work.

Would it be possible to also be able to get auto completion here?

```
@Component({
  selector: 'app-button',
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  templateUrl: './app-button.component.html',
  host: {
    class: 'block bg-white rounded-sm' // this is where I want auto completion
  }
})
export class ButtonComponent {}
```

**Comment by wongjn** [comment]
You can look at using the `tailwindCSS.experimental.classRegex` setting. Since it is experimental, there's not much documentation on it, but an internet search should hopefully get you going to set it up for your use case.

**Comment by leon** [comment]
For those of you looking at this later on this is what worked for me.

Add this to your settings.json

```
"tailwindCSS.experimental.classRegex": ["class:\\s'([^`]*)'"]
```

**Reply by SeraphCoding** [reply]
2 year old random answer on a small thread - exactly what I was looking for. Thank you so much!

**Comment by HitomiTenshi** [comment]
@leon's regex has one problem: Prettier adds a newline between `class:` and the string when the class list is very long, which the regex doesn't catch.

Additionally, Angular also allows to [dynamically add a single class](https://angular.dev/guide/components/host-elements#binding-to-the-host-element) with the `'[class.my-class]': 'myBooleanValue'` syntax.

I updated the regex to support tailwind's autocomplete for both cases:

```json
"tailwindCSS.experimental.classRegex": ["class:\\s*'([^']*)", "'\\[class\\.([^\\]\\s']*)"]
```

**Comment by HyperLife1119** [comment]
This is my configuration, covering some common usages, welcome to add :)

```json5
"tailwindCSS.experimental.classRegex": [
    "class:\\s'(.*?)'",         // class: 'xxx'
    "\\[class\\.(.*?)]",        // [class.xxx]="condition"
    "\\[class]':\\s.*?'(.*?)'", // '[class]': `condition ? 'xxx' : ...`
    "\\[class]':\\s.+'(.*?)'",  // '[class]': `condition ? ... : 'xxx'`
],
```

**Reply by luangong** [reply]
This is what we use currently:

```jsonc
"tailwindCSS.experimental.classRegex": [
  "class: ['`]([^'`]*)['`]",                    // class: 'foo bar' or class: `foo bar`
  ["'\\[class\\]': '([^']*)'", "\"([^\"]*)\""], // '[class]': 'foo ? "bar" : "baz"'
  ["'\\[class\\]': `([^`]*)`", "'([^']*)'"],    // '[class]': `foo ? 'bar' : 'baz'`
  "\\[class\\.([^\\]]*)\\]",                    // '[class.foo]': 'bar' or [class.foo]="bar"
  "'animate\\.(?:enter|leave)': '([^']*)'",     // 'animate.enter': 'foo bar' or 'animate.leave': 'foo bar'
],
```

References:

- [How to get Tailwind IntelliSense anywhere](https://dev.to/tiu/how-to-get-tailwind-intellisense-anywhere-53ej)
- [Tailwind CSS: Autocomplete Done Right](https://www.sixian.li/writing/tailwind-autocomplete-done-right)
- [paolotiu/tailwind-intellisense-regex-list · GitHub](https://github.com/paolotiu/tailwind-intellisense-regex-list)
- [tailwindlabs/tailwindcss-intellisense/packages/tailwindcss-language-service/src/util/classes.ts](https://github.com/tailwindlabs/tailwindcss-intellisense/blob/main/packages/tailwindcss-language-service/src/util/classes.ts)

**Comment by rozsazoltan** [comment]
Reference:
* [Displaying color boxes next to the TailwindCSS class names indicating the color, anywhere such as in class or className attributes](https://stackoverflow.com/a/79561483/15167500)
* [VSCode Tailwind CSS Intellisense autocompletion and suggestions not working after installing Tailwind v4](https://stackoverflow.com/a/79433860/15167500)
